src/vm/hints/find_element_hint.rs

```diff
@@ -0,0 +1,558 @@
+            )
+            .expect("Unexpected memory insert fail");
+
+        vm.references = HashMap::from([
+            (
+                0,
+                HintReference {
+                    register: Register::FP,
+                    offset1: -4,
+                    offset2: 0,
+                    inner_dereference: false,
+                    ap_tracking_data: None,
+                },
+            ),
+            (
+                1,
+                HintReference {
+                    register: Register::FP,
+                    offset1: -3,
+                    offset2: 0,
+                    inner_dereference: false,
+                    ap_tracking_data: None,
+                },
+            ),
+            (
+                2,
+                HintReference {
+                    register: Register::FP,
+                    offset1: -2,
+                    offset2: 0,
+                    inner_dereference: false,
+                    ap_tracking_data: None,
+                },
+            ),
+            (
+                3,
+                HintReference {
+                    register: Register::FP,
+                    offset1: -1,
+                    offset2: 0,
+                    inner_dereference: false,
+                    ap_tracking_data: None,
+                },
+            ),
+            (
+                4,
+                HintReference {
+                    register: Register::FP,
+                    offset1: 0,
+                    offset2: 0,
+                    inner_dereference: false,
+                    ap_tracking_data: None,
+                },
+            ),
+        ]);
+
+        let mut ids = HashMap::<String, BigInt>::new();
+        for (i, s) in ["array_ptr", "elm_size", "n_elms", "index", "key"]
+            .iter()
+            .enumerate()
+        {
+            ids.insert(s.to_string(), bigint!(i as i32));
+        }
+
+        (vm, ids)
+    }
+
+    #[test]
+    fn element_found_by_search() {
+        let (mut vm, ids) = init_vm_ids();
+
+        assert_eq!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Ok(())
+        );
+
+        assert_eq!(
+            vm.memory.get(&MaybeRelocatable::from((0, 3))),
+            Ok(Some(&MaybeRelocatable::Int(bigint!(1))))
+        )
+    }
+
+    #[test]
+    fn element_found_by_global() {
+        let (mut vm, ids) = init_vm_ids();
+        vm.find_element_index = Some(bigint!(1));
+
+        assert_eq!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Ok(())
+        );
+
+        assert_eq!(
+            vm.memory.get(&MaybeRelocatable::from((0, 3))),
+            Ok(Some(&MaybeRelocatable::Int(bigint!(1))))
+        )
+    }
+
+    #[test]
+    fn element_not_found_search() {
+        let (mut vm, ids) = init_vm_ids();
+        vm.memory.data[0][4] = Some(MaybeRelocatable::from(bigint!(7)));
```

Comment:
Try not to access `memory.data` directly. We want to abstract that.

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/294/commits/44039cc5f34fe872d57d50c8bf268eed77b0fc4c